src/litdata/streaming/config.py

```diff
@@ -223,3 +241,26 @@ def _validate_item_loader(self) -> None:
             and not isinstance(self._item_loader, TokensLoader)
         ):
             raise ValueError("Please, use Cache(..., item_loader=TokensLoader(block_size=...))")
+
+
+def load_subsampled_chunks(subsampled_files: List[str], original_chunks: List[Dict[str, Any]]) -> List[Dict[str, Any]]:
+    """Loads Chunks based on the basis of subsample provided."""
+    my_subsampled_chunks: List[Dict[str, Any]] = [{}] * len(subsampled_files)
+
+    assert len(my_subsampled_chunks) == len(subsampled_files)
+
+    for curr_chunk in original_chunks:
+        if curr_chunk["filename"] in subsampled_files:
+            idx = subsampled_files.index(curr_chunk["filename"])
+            my_subsampled_chunks[idx] = curr_chunk
+
+    # if any idx of my_subsampled_chunks is None, means,
+    # some elements in subsampled_files were not actually part of chunks
+    # raise error
+    if any(not _my_subsampled_chunk for _my_subsampled_chunk in my_subsampled_chunks):
+        raise ValueError(
+            "Mismatch in subsampled files and the chunks loaded",
+            "Make sure subsampled chunks are actually part of the original chunk",
+        )
+
+    return my_subsampled_chunks
```

Comment:
No need to put `my_` in front of all variables. Let's remove those.

Reply:
Sure.